pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -736,8 +769,101 @@ func (r *resourceManager) emptyPodDisruptionBudget() *policyv1beta1.PodDisruptio
 	return &policyv1beta1.PodDisruptionBudget{ObjectMeta: metav1.ObjectMeta{Name: "gardener-resource-manager", Namespace: r.namespace}}
 }
 
+func (r *resourceManager) ensureMutatingWebhookConfiguration(ctx context.Context) error {
+	mutatingWebhookConfiguration := r.emptyMutatingWebhookConfiguration()
+
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.client, mutatingWebhookConfiguration, func() error {
+		mutatingWebhookConfiguration.Labels = appLabel()
+		mutatingWebhookConfiguration.Webhooks = r.getMutatingWebhookConfigurationWebhooks()
+		return nil
+	})
+	return err
+}
+
+func (r *resourceManager) emptyMutatingWebhookConfiguration() *admissionregistrationv1.MutatingWebhookConfiguration {
+	suffix := ""
+	if r.values.TargetDiffersFromSourceCluster {
+		suffix = "-shoot"
+	}
+	return &admissionregistrationv1.MutatingWebhookConfiguration{ObjectMeta: metav1.ObjectMeta{Name: "gardener-resource-manager" + suffix, Namespace: r.namespace}}
+}
+
+func (r *resourceManager) ensureShootResources(ctx context.Context) error {
+	var (
+		registry = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+
+		mutatingWebhookConfiguration = r.emptyMutatingWebhookConfiguration()
+	)
+
+	mutatingWebhookConfiguration.Labels = appLabel()
+	mutatingWebhookConfiguration.Webhooks = r.getMutatingWebhookConfigurationWebhooks()
+
+	data, err := registry.AddAllAndSerialize(mutatingWebhookConfiguration)
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForShoot(ctx, r.client, r.namespace, ManagedResourceName, false, data)
+}
+
+func (r *resourceManager) getMutatingWebhookConfigurationWebhooks() []admissionregistrationv1.MutatingWebhook {
```

Comment:
It's probably a good idea to make the webhook config reusable between the component and envtest package. E.g. like it is done for gsac:
https://github.com/gardener/gardener/blob/aeffd6154d2cee2c84b36530d812d9d3a4ef59f4/pkg/operation/botanist/component/seedadmissioncontroller/seedadmissioncontroller.go#L330